<?xml version="1.0"?>

<launch>

   <!--Launch the world with the robots on Gazebo  -->	
   <include file="$(find usar_gazebo)/launch/landslide.launch"/>
   <!--<include file="$(find usar_gazebo)/launch/post-disaster.launch"/> -->

<!-- LAUNCH ROBOTS -->

  <!-- Launch Pioneer3AT 1 -->
  <group ns="pioneer3at_1">
    <include file="$(find pioneer3at_description)/launch/pioneer3at.launch">
      <arg name="robot_name" value="pioneer3at_1"/>
      <arg name="xcoordinate" value="-33.0"/>
      <arg name="ycoordinate" value="-30.0"/>
      <arg name="zcoordinate" value="0.45"/>
      <arg name="roll" value="0.0"/>
      <arg name="pitch" value="0.0"/>
      <arg name="yaw" value="0.0"/>
    </include>
  </group> 

  <!-- Launch Pioneer3AT 2 
  <group ns="pioneer3at_2">
    <include file="$(find pioneer3at_description)/launch/pioneer3at.launch">
      <arg name="robot_name" value="pioneer3at_2"/>
      <arg name="xcoordinate" value="-33.0"/>
      <arg name="ycoordinate" value="-31.0"/>
      <arg name="zcoordinate" value="0.45"/>
      <arg name="roll" value="0.0"/>
      <arg name="pitch" value="0.0"/>
      <arg name="yaw" value="0.0"/>
    </include>
  </group> -->

  <!-- Spawn simulated quadrotor uav 1
  <group ns="UAV_1">
    <include file="$(find hector_quadrotor_gazebo)/launch/quadrotor.launch" >
      <arg name="name" default="UAV_1"/>
      <arg name="tf_prefix" value="UAV_1" />
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
      <arg name="controllers" value="
    	  controller/position
    	  controller/velocity
    	  controller/attitude
          "/>
      <arg name="x" default="-33.0"/>
      <arg name="y" default="-33.0"/>
      <arg name="z" default="0.6"/> 
    </include>
  </group> -->

  <!-- Spawn simulated quadrotor uav 2
  <group ns="UAV_2">
    <include file="$(find hector_quadrotor_gazebo)/launch/quadrotor.launch" > 
      <arg name="name" default="UAV_2"/>
      <arg name="tf_prefix" value="UAV_2" />
      <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/>
      <arg name="controllers" value="
    	  controller/position
    	  controller/velocity
    	  controller/attitude
          "/>
      <arg name="x" default="-33.0"/>
      <arg name="y" default="-34.0"/>
      <arg name="z" default="0.6"/> 
    </include>
  </group>  -->

  <!-- Launch Pioneer2DX 1 
  <group ns="pioneer2dx_1">
    <include file="$(find pioneer2dx_ros)/launch/pioneer2dx.launch">
      <arg name="robot_name" value="pioneer2dx_1"/>
      <arg name="xcoordinate" value="0.0"/>
      <arg name="ycoordinate" value="7.0"/>
      <arg name="zcoordinate" value="1.6"/>
      <arg name="roll" value="0.0"/>
      <arg name="pitch" value="0.0"/>
      <arg name="yaw" value="0.0"/>
    </include>
  </group> -->

  <!-- Launch Pioneer2DX 2 
  <group ns="pioneer2dx_2">
    <include file="$(find pioneer2dx_ros)/launch/pioneer2dx.launch">
      <arg name="robot_name" value="pioneer2dx_2"/>
      <arg name="xcoordinate" value="0.0"/>
      <arg name="ycoordinate" value="8.0"/>
      <arg name="zcoordinate" value="1.6"/>
      <arg name="roll" value="0.0"/>
      <arg name="pitch" value="0.0"/>
      <arg name="yaw" value="0.0"/>
    </include>
  </group> -->

<!-- LAUNCH HUMAN TEAMS -->

  <!-- Launch fireman1 
  <group ns="human1">
    <include file="$(find human_teams)/launch/fireman1.launch">
      <arg name="human_name" value="human1"/>
      <arg name="xcoordinate" value="-34.0"/>
      <arg name="ycoordinate" value="-34.0"/>
      <arg name="zcoordinate" value="0.18"/>
      <arg name="yaw" value="1.57"/>
    </include>
  </group> -->

  <!-- Launch fireman2 
  <group ns="human2">
    <include file="$(find human_teams)/launch/fireman2.launch">
      <arg name="human_name" value="human2"/>
      <arg name="xcoordinate" value="-34.0"/>
      <arg name="ycoordinate" value="-33.0"/>
      <arg name="zcoordinate" value="0.18"/>
      <arg name="yaw" value="1.57"/>
    </include>
  </group> -->

  
<!--LAUNCH MAP_MERGE
  <include file="$(find pioneer3at_slam)/launch/multi_map_merge.launch">
      <arg name="namespace" value="pioneer3at"/>
      <arg name="first_robot"  value="pioneer3at_1"/>
      <arg name="second_robot" value="pioneer3at_2"/>

      <arg name="first_robot_x_pos" value=" -33.0"/>
      <arg name="first_robot_y_pos" value=" -30.0"/>
      <arg name="first_robot_z_pos" value=" 0.45"/>
      <arg name="first_robot_yaw"   value=" 0.0"/>

      <arg name="second_robot_x_pos" value="-33.0"/>
      <arg name="second_robot_y_pos" value="-31.0"/>
      <arg name="second_robot_z_pos" value=" 0.45"/>
      <arg name="second_robot_yaw"   value=" 0.0"/> 
  </include> -->

<!-- LAUNCH RVIZ -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find full_system)/rviz/multi_slam.rviz"/> 

  <master auto="start"/>
  <param name="/use_sim_time" value="true"/> 

<!-- LAUNCH RQT
  <node name="rqt_topic" pkg="rqt_topic" type="rqt_topic" output="screen" /> 
  <node name="rqt_graph" pkg="rqt_graph" type="rqt_graph" output="screen" /> 
  <node name="rqt_console" pkg="rqt_console" type="rqt_console" output="screen" /> 
  <node name="rqt_tf_tree" pkg="rqt_tf_tree" type="rqt_tf_tree" output="screen" /> -->


</launch>
